ngraph/frontend/onnx_import/src/editor/editor.cpp

```diff
@@ -369,3 +403,118 @@ void onnx_import::ONNXModelEditor::set_input_values(
         modify_initializer(*onnx_initializer, name, values, onnx_input);
     }
 }
+
+void onnx_import::ONNXModelEditor::ONNXModelEditor::replace_nodes(
+    std::vector<std::vector<int>> node_indexes, Operator node_generator)
+{
+    std::string new_op_name = "custom_op_" + std::to_string(m_custom_op_ID++);
+
+    shift_node_indexes(node_indexes);
+
+    for (auto node_vector : node_indexes)
+    {
+        auto new_end = std::unique(node_vector.begin(), node_vector.end());
+        node_vector.erase(new_end, node_vector.end());
+        replace_nodes(node_vector, node_generator, new_op_name);
+    }
+    onnx_import::register_operator(new_op_name, 1, "org.openvinotoolkit.editor", node_generator);
+}
+
+void onnx_import::ONNXModelEditor::ONNXModelEditor::replace_nodes(
+    std::vector<int> node_indexes, onnx_import::Operator node_generator, std::string new_op_name)
+{
+    std::vector<std::string> inputs;
+    std::vector<std::string> outputs;
+
+    auto nodes = model().graph().node();
+
+    // 1. create a list of inputs and outputs of new custom op.
+    for (auto node_index : node_indexes)
+    {
+        inputs.insert(
+            inputs.end(), nodes[node_index].input().begin(), nodes[node_index].input().end());
+        outputs.insert(
+            outputs.end(), nodes[node_index].output().begin(), nodes[node_index].output().end());
+    }
+
+    // 2. get internal tensors via intersection of inputs and outputs.
+    std::vector<std::string> internal_tensors;
+
+    std::set<std::string> inputs_set(inputs.begin(), inputs.end());
+    std::set<std::string> outputs_set(outputs.begin(), outputs.end());
+
+    std::set_intersection(inputs_set.begin(),
+                          inputs_set.end(),
+                          outputs_set.begin(),
+                          outputs_set.end(),
+                          std::inserter(internal_tensors, internal_tensors.end()));
+
+    auto is_external_tensor = [&node_indexes, &nodes](std::string tensor_name) {
+        bool is_external = false;
+        for (int i = 0; i < nodes.size(); ++i)
+        {
+            if (std::count(node_indexes.begin(), node_indexes.end(), i))
```

Comment:
I think you should break the loop when you figure out that the tensor is external. I suggest to put this inside the for-loop:

```
const bool in_indices = std::any_of(node_indexes.begin(), node_indexes.end(), i);
const bool is_external = std::any_of(nodes[i].input().begin(), nodes[i].input().end(), tensor_name);

if (!in_indices && is_external)
    return true;
```

and a simple `return false` after the loop.